packages/edit-site/src/components/sidebar-dataviews/use-data-views.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * WordPress dependencies
+ */
+import { useEntityRecords } from '@wordpress/core-data';
+import { useMemo } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { OPERATOR_IS_ANY, OPERATOR_IS_NONE } from '../../utils/constants';
+import { DEFAULT_VIEWS } from './default-views';
+
+export default function useDataViews( postType ) {
+	// Fetch all the posts of the post type.
+	const data = useEntityRecords( 'postType', postType, {
+		per_page: -1,
```

Comment:
I am a little skeptical about fetching all the results using -1 like this.
But I am not able to think about any better way. Please suggest if you get any better approach. 🙏

Reply:
If you just want the total count, did you try looking at `totalItems` return by [useEntityRecords](https://github.com/WordPress/gutenberg/blob/2139bc2ceb6101537be196ef3fddc965b90a6b75/packages/core-data/src/hooks/use-entity-records.ts#L86-L86)?

Example:

```js
const { totalItems } = useEntityRecords( 'postType', 'post', { status: [ 'any', 'trash' ] } ) );
```

`useEntityRecords` makes a call to the `getEntityRecordsTotalItems` selector, which returns the count sent via [response headers in X-WP-Total when calling getEntityRecords](https://github.com/WordPress/gutenberg/blob/2139bc2ceb6101537be196ef3fddc965b90a6b75/packages/core-data/src/resolvers.js#L238-L238).

Or you call `getEntityRecordsTotalItems` directly maybe.

Reply:
Trying something over here with a new route and [wp_count_posts](https://developer.wordpress.org/reference/functions/wp_count_posts/)

- https://github.com/WordPress/gutenberg/pull/66294